core-js/src/test/javascript/cdf/components/UnmanagedComponent-spec.js

```diff
@@ -219,10 +222,85 @@ define([
     });
 
     /**
+     * ## Unmanaged Component # should allow changes in the datasources
+     */
+    it("should allow changes in the datasources", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      uquery.preExecution = function () {
+        this.queryDefinition.dataSource = "otherDatasource";
+      };
+      uquery._setQuery = function (queryDef, queryOptions) {
+        expect(queryDef.dataSource).toBe("otherDatasource");
+        done();
+      };
+      dashboard.update(uquery);
+    });
+
+    /**
+     * ## Unmanaged Component # should allow changes in the datasources by override
+     */
+    it("should allow changes  in the datasources by override", function(done) {
```

Comment:
Use a more generic description. E.g.: "preExecution should allow overriding the query definition object"